docs/references/restatectl.mdx

```diff
@@ -198,3 +202,62 @@ Alive partition processors (nodes config v21, partition table v21)
+
+New log servers come up in the `provisioning` state and will automatically transition to `read-write`. The `read-write` state means that the node is considered both healthy to read from and accept writes, that is it may be selected as a node set member for new loglet segments.
+
+### list-servers
+
+You can view the current storage state of log servers in your cluster using the `list-servers` sub-command.
+
+```shell
+restatectl replicated-loglet list-servers
+```
+
+<details className="grey-details">
+    <summary>Output</summary>
+```
+Node configuration v12
+Log chain v3
+ NODE  GEN   STORAGE-STATE  HISTORICAL LOGLETS  ACTIVE LOGLETS
+ N1    N1:3  read-write     4                   2
+ N2    N2:2  read-write     4                   2
+ N3    N3:2  read-write     4                   2
+```
+</details>
+
+Other valid storage include `data-loss`, `read-only`, and `disabled`. Nodes may transition to `data-loss` if they detect that some previously written data is not available. This does not necessarily imply corruption, only that such nodes may not participate in some quorum checks. Such nodes may transition back to `read-write` if they can be repaired.
+
+The `read-only` and `disabled` states are of particular interest to operators. Log servers in `read-only` storage state may continue to serve both reads and writes, but will no longer be selected as participants in new segments' node sets. The control plane will reconfigure existing logs to move away from such nodes.
+
+### set-storage-state
+
+<Admonition type="caution" title="Danger of data loss">
+    `set-storage-state` is a low-level command that allows you to directly set log servers' storage-state. Changing this can lead to cluster unavailability or data loss.
+</Admonition>
+
+Use the `set-storage-state` sub-command to manually set the state, for example to drain log servers. Consider the following example:
+
+```shell
+restatectl replicated-loglet set-storage-state --node-id 1 --storage-state read-only
+```
+
+<details className="grey-details">
+    <summary>Output</summary>
+```
+Node N1 storage-state updated from read-write to read-only
+```
+</details>
+
+The cluster controller reconfigures the log node set to exclude `N1`. Depending on the configured log replication level, you may see a warning about compromised availability or, if insufficient log servers are available to achieve the minimum required replication, the log will stop accepting writes altogether. You have to take care as `restatectl` does not currently check whether the cluster will be able to generate new node sets with the remaining log servers.
+
+The `disabled` storage state tells Restate that a log server must not be considered for reads or writes - effectively it is no longer a part of the replicated log. It is only safe to transition a `read-only` server to the `disabled` state once the node has been removed from all node sets and any log chain segments that reference it have been trimmed.
```

Comment:
This is not supported yet.

Reply:
I saw that from the code but felt it was worth putting in anyway. Are you happy if I leave it in with a note that this is not yet supported?

Reply:
I would probably add it once we support it.